### DISCUSSION TITLE
How to reference files from the site root, not tenant

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#10081] [Q&A]
I build a multi-tenant app and use a server-side Blazor.
The component is rendered into some tenant view but references files from `wwwroot` of the main site.
How can I instruct it not to add tenant name into URL or forward site root `wwwroot` to each tenant?

**Comment by Skrypt** [comment]
https://docs.orchardcore.net/en/latest/docs/reference/modules/Tenants/index.html#static-file-provider-feature

**Reply by xperiandri** [reply]
Do you mean that placing a file in app project's `wwwroot` must serve it for each tenant automatically?

**Reply by maglini** [reply]
@Skrypt We cant achieve resource "blazor.server.js" 
Using in _Layout.cshtml: 
```
<script src="~/Bitrix24.Tenant/_framework/blazor.server.js"></script>
<script src="~/_framework/blazor.server.js"></script>
```
We can receive that only from root by next URL: localhost:44318/_framework/blazor.server.js

Any ideas to fix ?

**Reply by xperiandri** [reply]
But how to have a single file for all tenants?

**Reply by xperiandri** [reply]
Embed files and do like I did here?
https://github.com/OrchardCMS/OrchardCore/issues/2966#issuecomment-897105239

**Comment by ns8482e** [comment]
Do not use `UseStaticFiles` in host project.

**Comment by ns8482e** [comment]
@xperiandri PR #10118 solve this.

**Reply by xperiandri** [reply]
So how must my code change? Just remove static files provider for Blazor assembly?

**Reply by ns8482e** [reply]
@xperiandri  currently, OC tenant pipeline replaces, existing FileProviders,   In the PR #10118 - instead of replacing it combines existing FileProviders - so if PR is accepted and merged,  tenant pipeline will  also resolve File Provider from  Blazor Server or any other external library that uses `PostConfigure` for staticfile options..

So, then after the PR #10118 ,  all you need in Startup is, 

```cs
services.AddOrchardCore()
                .AddMvc()
                .WithTenants()
                .ConfigureServices(svc =>
                {
                   // Configure Blazor in tenant pipeline
                    svc.AddServerSideBlazor();
                    svc.AddSingleton<WeatherForecastService>();
                }).Configure((app, routes, serviceProvider) =>
                {
                    routes.MapBlazorHub();
                    routes.MapFallbackToAreaPage("/_Host", "Mvc.Web");
                });
```

If you need to resolve `wwwroot` from host to a tenant then you need following configuration

```json
"CustomerA": {
      "State": "Running",
      "RequestUrlHost": null,
      "RequestUrlPrefix": "customer-a",
      "Features": [],
      "CustomTitle": "Customer A",
      "CustomSetting": "Custom setting for Customer A",
      "StaticFileOptions": {
        "ServeHostWebRoot": true
      }
    }
```

**Reply by xperiandri** [reply]
Perfect! Thanks!
